### PR TITLE
[py] An expectation for checking if non-empty text is present in element

### PR DESCRIPTION
**Pull request by wwakabobik**
text_to_be_present_in_element_non_empty method is added to check that
the web element contains any non-empty text. Usually it's needed when
text is generated on-the-fly on the server side, and of course, that
the text of element is not empty.

Also, added self-test for this feature.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: py/docs/source/webdriver_support/selenium.webdriver.support.expected_conditions.rst [hunk 1]
```diff
@@ -34,6 +34,7 @@
       text_to_be_present_in_element
       text_to_be_present_in_element_attribute
       text_to_be_present_in_element_value
+      text-to_be_present_in_element_non_empty
```
Comment (symonk) [inline py/docs/source/webdriver_support/selenium.webdriver.support.expected_conditions.rst]: typo `-` here

File: py/selenium/webdriver/support/expected_conditions.py [hunk 2]
```diff
@@ -254,6 +254,20 @@ def _predicate(driver):
     return _predicate
 
 
+def text_to_be_present_in_element_non_empty(locator):
```
Comment (symonk) [inline py/selenium/webdriver/support/expected_conditions.py:257]: I need to discuss with those in the core team; I am not sure our current stance on adding additional expected_conditions - and we should update the docs to be clear (if we prefer them to be more user defined in their own projects) - will report back shortly

**Comment by symonk** [comment]
Thanks a lot for the contribution @wwakabobik - I'm quite new here so just need to figure out with folks what our current stance is on expected_conditions (additional ones living in selenium core; will let you know as soon as I know!)

File: py/docs/source/webdriver_support/selenium.webdriver.support.expected_conditions.rst (hunk 1 above)
Comment (wwakabobik) [reply py/docs/source/webdriver_support/selenium.webdriver.support.expected_conditions.rst]: thanks, fixed

File: py/selenium/webdriver/support/expected_conditions.py (hunk 2 above)
Comment (wwakabobik) [reply py/selenium/webdriver/support/expected_conditions.py:257]: Ok, thanks, I'll wait for decision

**Comment by titusfortner** [comment]
Thanks for your contribution.

.NET got rid of the expected conditions classes entirely because everyone wanted to add their own (e.g., now make one for text with whitespace), and the maintenance burden was too great. So now .NET & Ruby both require all waits to be done via lambdas. 

Selenium users in the other languages should make more use of lambdas. For this one, Python is especially nice because an empty string is "falsy":

```python
WebDriverWait(driver, 3).until(lambda d: d.find_element(locator).text.strip())
```

**Pull request closed without merging**